src/lib.rs

```diff
@@ -520,6 +534,39 @@ impl<T> SliceDeque<T> {
         }
     }
 
+    /// Returns the slice of uninitialized memory between the `tail` and the
+    /// `head`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[macro_use] extern crate slice_deque;
+    /// # fn main() {
+    /// let mut d = sdeq![1,2,3];
+    /// let cap = d.capacity();
+    /// let len = d.len();
+    /// unsafe {
+    ///     {
+    ///         // This slice contains the uninitialized elements in
+    ///         // the deque:
+    ///         let mut s = d.uninitialized_mut_slice();
+    ///         assert_eq!(s.len(), cap - len);
+    ///         // We can write to them and for example bump the tail of
+    ///         // the deque:
+    ///         s[0] = 4;
+    ///         s[1] = 5;
+    ///     }
+    ///     d.move_tail(2);
+    /// }
+    /// assert_eq!(d, sdeq![1,2,3,4,5]);
+    /// # }
+    /// ```
+    pub unsafe fn uninitialized_mut_slice(&mut self) -> &mut [T] {
```

Comment:
This could be shortened to `uninit_mut_slice()` unless you prefer having it spelled out.

I was trying to think of a more descriptive name for the unused space between the tail and the head but I couldn't think of any so I guess this works.

Reply:
> This could be shortened to uninit_mut_slice() unless you prefer having it spelled out. 

I think I prefer it spelled out for now.

> Though, I think the mut is typically placed at the end of the method name.

I thought so too but it seems that the `VecDeque` method `as_mut_slice` spells the `mut` in the middle. In any case, it makes no sense to get a read only slice to the uninitialized region, so I will just kill the `mut`.

> I was trying to think of a more descriptive name for the unused space between the tail and the head but I couldn't think of any so I guess this works.

Maybe `get_uninitialized_slice` ? `uninitialized_slice` ? I am bad at names.

Reply:
I'm mostly hung up on the word "uninitialized". It's long and isn't the best at describing what's actually being returned here. "uninitialized" implies the section of the buffer that's *never* been written to and thus may contain junk data from previous allocations; it implies that, if the queue is filled to capacity (without reallocating) and then emptied, the slice it returns will be 0-length because there will be no uninitialized indices left, just ones that have been moved out of.

I want to say "uninhabited" instead but it's just as long and "uninhabited" has a different meaning in Rust; it describes a type like `enum Void { }`, which can't exist at runtime. And "empty slice" is confusing too as it usually means `&[ ]` or `&mut [ ]`.

What about "free" or "unused" or "vacant"? Or something using more specific terminology like `tail_head_slice`? I can't find any specific term for this space in the academic descriptions of deques, I guess because it's more of an implementation detail than something pertaining to the abstract datastructure itself. Maybe I'm being too pedantic, I don't know.

Reply:
> "uninitialized" implies the section of the buffer that's never been written to and thus may contain junk data from previous allocations; 

I see. I meant it to denote that the memory of the slice is in a `mem::uninitialized()` state, but I can see how this is confusing.

> tail_head_slice

For the lack of a better name, that's pretty good.

What about "remainder" ? To denote the remaining elements that are not part of the deque?

Reply:
> What about "remainder" ? To denote the remaining elements that are not part of the deque?

That implies the elements after the head, I think. Like the tail of a list in functional programming parlance.

What about `exterior_slice`, describing the area of the buffer that's exterior to the deque? Too obscure?